src/job_cache/job_cache.cpp

```diff
@@ -215,49 +224,81 @@ FindJobResponse Cache::read(const FindJobRequest &find_request) {
   MessageParser parser(socket_fd.get());
   std::vector<std::string> messages;
 
-  while (true) {
-    MessageParserState state = parser.read_messages(messages);
+  MessageParserState state = MessageParserState::Continue;
+  do {
+    state = parser.read_messages(messages);
 
     if (state == MessageParserState::StopFail) {
-      // TODO: Try to reconnect to the daemon, launching our own if need be.
-      //       if that fails then depending on user preference either fail
-      //       or return a cache miss.
-      wcl::log::fatal("Cache::read(): failed receiving message");
+      // TODO: Add config var to determine if fail is a cache miss
+      if (false) {
+        return wcl::result_value<FindJobError>(FindJobResponse(wcl::optional<MatchingJob>{}));
+      }
+
+      wcl::log::error("Cache::read(): failed receiving message");
```

Comment:
These non-fatal errors should be shown to the human while running so that they know what could be causing the slow down...and they should possibly be warnings I'm not sure.

Reply:
I still don't agree that *all* `log::error`s should be displayed to the user. Which is why we need the filtered logger in https://github.com/sifive/wake/pull/1284.

Can we punt this comment until that lands?

Reply:
We can punt but I think if this message comes up and user's have the config flag enabled to not fail they need to see these error messages.